### PR TITLE
Remove string allocation from Uri.PathDifference

### PR DESCRIPTION
**Pull request by stephentoub**
cc: @rmkerr

### CODE REVIEW

**Comment by danmosemsft** [comment]
@stephentoub did you just spot this or are you preparing some analyzer that looks for things like appending temp strings to STringBuilders?

**Comment by stephentoub** [comment]
> analyzer that looks for things like appending temp strings to STringBuilders?

We effectively already have one: source.dot.net.  I've just been looking through the results of things calling StringBuilder.Append(object), StringBuilder.ToString(), etc. (Unfortunately source.dot.net is woefully out of date at the moment, but @alexperovich is working on fixing it, hopefully soon.)

File: src/System.Private.Uri/src/System/Uri.cs [hunk 1]
```diff
@@ -5367,7 +5367,7 @@ private static string PathDifference(string path1, string path2, bool compareCas
             // Same path except that path1 ended with a file name and path2 didn't
             if (relPath.Length == 0 && path2.Length - 1 == si)
                 return "./"; // Truncate the file name
-            return relPath.ToString() + path2.Substring(si + 1);
+            return relPath.Append(path2.AsSpan(si + 1)).ToString();
```
Comment (bartonjs) [inline src/System.Private.Uri/src/System/Uri.cs]: Is there a significant difference between Substring() and AsSpan().ToString()?

Comment (stephentoub) [reply src/System.Private.Uri/src/System/Uri.cs]: No... but this isn't AsSpan().ToString.  I think you're missing one of the parens in there :)

Comment (bartonjs) [reply src/System.Private.Uri/src/System/Uri.cs]: Ooooh.  I was.  D'oh.

**Pull request merged**